org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/desktop/DashboardController.java

```diff
@@ -436,6 +436,15 @@ public  boolean render(Component content, MDashboardContent dc, DashboardRunnabl
 				btn.setLabel(Msg.getMsg(Env.getCtx(), "ViewReportInNewTab"));
 				toolbar.appendChild(new Separator("vertical"));
 				toolbar.appendChild(btn);
+
+				btn = new ToolBarButton();
+				btn.setAttribute("Refresh", true);
+				btn.setAttribute("AD_Process_ID", AD_Process_ID);
+				btn.setAttribute("ProcessParameters", processParameters);
+				btn.setAttribute("Parent", content);
+				btn.setImage(ThemeManager.getThemeResource("images/Refresh16.png"));
+				btn.addEventListener(Events.ON_CLICK, this);
```

Comment:
use btn.addEventListener(Events.ON_CLICK, e -> embedReport(content, AD_Process_ID, processParameters, false)); and you can remove many of your changes :)